1_cpp_basics/Assignment_1.md

```diff
@@ -0,0 +1,63 @@
+# Pixels Assignment 1
+
+In this assignment you will peactise problems based on the concepts of arrays, pointers, relationship of arrays and pointers and vectors.
+
+## Basic of C++
+
+ Fun with pointers.
+
+**Q1** A teacher awards each student with 3 candies at the start of the day. Each time a student misbehaves she takes away 2 candies from them and awards them with an extra candy when they perform well in their tests. There are 4 students in the class.
+WAP to help the teacher distribute the right number of candies.
+
+**Expected Output**
+
+[![Screenshot-from-2023-03-27-22-11-37.png](https://i.postimg.cc/d1K8n1f7/Screenshot-from-2023-03-27-22-11-37.png)](https://postimg.cc/CZmBxFSY)
+
+*Hint U can use the recently taught concepts of pointer arithmetics*
+
+Q2  A shopkeeper keeps a record of his inventory. Write a program to help him change the stock as and when the orders are placed. Too easy right 😉.Let's make things more interesting 😀. Use the concept of pointers to access this two-dimensional array.
+
+
+**Output is similar to that in Q1**
+
+
+| Unique ID of good |Price | In Stock  |
+| -------- | -------- | -------- |
+|0001      |250       |10        |
+|0002      |500       |20        |
+|0003      |1000      |30        |
+|0004      |1500      |40        |
+
+*Hint an array is stored in a continuous block of memory 😉
+
+**Q3** Lets us build upon our knowledge of pointers and arrays even further. Create a Print function that prints out the entire two-dimensional array. (Pass a pointer array to the  function)
+
+**Sample Input**
+
+[![Screenshot-from-2023-03-27-22-20-09.png](https://i.postimg.cc/Vvzs7Q1B/Screenshot-from-2023-03-27-22-20-09.png)](https://postimg.cc/MMr8TLnv)
+
+**Sample Output**
+
+[![Screenshot-from-2023-03-27-22-20-20.png](https://i.postimg.cc/V6Y2pYr6/Screenshot-from-2023-03-27-22-20-20.png)](https://postimg.cc/McFsjJY2)
+
+****EXTRA SPECIAL BROWNIE POINTS Question****
+Write a similar function but print only the price of the items(use a pointer to a single element of the array)
+
+
+**Q4** We have learned the concepts of vectors now lets use them to render vectors which represent images
+
+
+** Hint we can represent black pixels by 0 and white ones by 1. lets say for a image with 5X5 size
+
+
+[![pic1.png](https://i.postimg.cc/k4RBx4Bz/pic1.png)](https://postimg.cc/wyHqz95Q)
+
+![](https://i.imgur.com/SeX8Ppg.png) 
+
+**Q5** Now copy the image into a new image and render a color inverted image wherein black is on top and white is at the bottom. 🤗
+
+![](https://i.imgur.com/CwUtTZb.png)
```

Comment:
Do not relate the question with images. since keep it within the scope of arrays